### PR TITLE
New Layers

### LINKED ISSUE #4701: cables, atmos pipes, and disposal pipes sprite layers inconsistency
## Description
image
cables and disposal pipes seem to disagree with layers, going up and down depending on the type of cable and pipe on the tile

red cables are unaffected by this

### LINKED ISSUE #9054: Square Cells doors access not set
Windoor cell doors on square have no access set

### PR DESCRIPTION
**Pull request by ThatDan123**
-Adds Electrical, Pipe and Disposal layers for each scene and moves the tiles from the underfloor layer to the appropriate layer.

CL: [Fix] Pipes and wires should now render on top of each other correctly

-Allows for work on tools like the T-ray scanner and for vent crawling showing the pipes by just changing the sorting order on the layer when necessary. #9061, #9062

-Mappers can just keep their changes and then add the new layers and then press the convert button on the Underfloor layer script if needed.

-Fixes #4701

-Fixes #9054

image

### CODE REVIEW

**Comment by ThatDan123** [comment]
For reviewing it the commits to SquareStation have all the script stuff in, though i think you can toggle to just view scripts somewhere on github

**Comment by ThatDan123** [comment]
Tests do pass locally is they fail to run again

**Comment by ThatDan123** [comment]
Only the labeller failing

File: UnityProject/Assets/Scripts/Tilemaps/Behaviours/InteractableTiles.cs [hunk 1]
```diff
@@ -251,33 +251,22 @@ public bool Interact(PositionalHandApply interaction)
...
-						var underFloorApply = new TileApply(interaction.Performer, interaction.UsedObject, interaction.Intent,
-							(Vector2Int) localPosition, this, underFloorTile, interaction.HandSlot, interaction.TargetPosition);
-
-						if (TryInteractWithTile(underFloorApply)) return true;
-					}
-				}
+				if (FindLayerInteraction(interaction, localPosition, LayerType.Underfloor)) return true;
+			}
+			// If the the tile is something that's supposed to be Electrical...
+			else if (basicTile.LayerType == LayerType.Electrical)
```
Comment (Bod9001) [inline UnityProject/Assets/Scripts/Tilemaps/Behaviours/InteractableTiles.cs]: would be better suited for a loop

**Review by Bod9001 — APPROVED** [review]

**Pull request merged**